.github/workflows/makefile.yml

```diff
@@ -0,0 +1,35 @@
+name: Makefile CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up OCaml
+      uses: ocaml/setup-ocaml@v2
+      with:
+        ocaml-compiler: 4.13.1
+        dune-cache: true
+
+    - name: Install OS deps
+      run: sudo apt update && sudo apt install -y libcurl4-gnutls-dev libevent-dev
+
+    - name: Install OCaml deps
+      run: opam install . --deps-only --with-test
+      
+    - name: Build
+      run: opam exec -- dune build
```

Comment:
would be better to call make but setup-ocaml doesn't support setting up opam env yet..

Reply:
> it doesn't look like tests are run..

They are run - It just isn't very verbose with it unless an error happens, like when trying to build before with Lwt_unix.yield which failed due to warns.

Reply:
omg, now it failed, wtf %)